src/lib/common/sem.cpp

```diff
@@ -294,55 +294,86 @@ int transSemGive(const char* who, const char* what)
 static pthread_mutex_t contexts_mutex = PTHREAD_MUTEX_INITIALIZER;
 static std::map<std::string, struct curl_context> contexts;
 
-//Statistics
+// Statistics
 static struct timespec accCCMutexTime = { 0, 0 };
 
 
-int get_curl_context(const std::string& key, struct curl_context *pcc)
+/* ****************************************************************************
+*
+* curl_context_cleanup - 
+*/
+void curl_context_cleanup(void)
```

Comment:
I also don't like the curl functions in this file.
Strictly sem (the name is sem.cpp ...)
All the curl functions should be in some other file, IMO.
Not really part of this PR though